### ISSUE TITLE
`InheritedWidget` does not support records on WASM

### ISSUE THREAD

**Issue by putnokiabel** [flutter/flutter#155510]
### Steps to reproduce

1. Create a new Flutter project with web support.
2. Run `flutter pub add provider`
3. Replace main.dart with the code sample below.
4. Run flutter `run -d chrome --wasm` to run the app on Chrome using WASM.
5. Note that the page cannot be built, and build 0 is printed to the console but build 1 is not.
6. Note that running the project with the exact same code works on web (e.g. `flutter run -d chrome`) as well as on Android and iOS but not on WASM.

### Expected results

The app runs without error and the provider is found when compiled to WASM (just like it works when compiled to any other platform).

### Actual results

When compiled to WASM, the provider is not found and an error is thrown.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

import 'package:provider/provider.dart';

typedef Record = ({String name});

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return Provider<Record>.value(
      value: (name: 'Hello, World!'),
      child: const MaterialApp(
        home: MyHomePage(),
      ),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key});

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    print('build 0');
    final record = context.read<Record>();
    print('build 1');

    return Scaffold(
      body: Center(
        child: Text(record.name),
      ),
    );
  }
}

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.3, on macOS 15.0 24A335 darwin-arm64, locale en-NL)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.2)
[✓] VS Code (version 1.93.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by putnokiabel** [comment]
Just an additional note that the owner of `provider` already concluded this is most likely a problem with the Flutter SDK, see https://github.com/rrousselGit/provider/issues/890

**Comment by mraleph** [comment]
cc @mkustermann @osa1

**Comment by osa1** [comment]
Fix in https://dart-review.googlesource.com/c/sdk/+/386200.

Thanks @mraleph for the help with debugging.

**Comment by putnokiabel** [comment]
Thank you for the quick action! 🙏😄

I'm not sure what the Dart SDK release schedule is like and how it overlaps with the Flutter SDK releases. Can I expect this fix to be in the next Flutter patch with hotfixes on `stable`?

**Comment by osa1** [comment]
@putnokiabel I created a cherry-pick request to the stable: https://github.com/dart-lang/sdk/issues/56817.

If accepted the fix should be in Dart 3.5.3. If not it will be a part of 3.6.

I think we make a Flutter release shortly after a minor SDK release, so hopefully it will be in Flutter release right after 3.5.3.

Closing this as the issue has been fixed.
